__tests__/app.test.js

```diff
@@ -104,6 +104,49 @@ describe("GET /api", () => {
                     expect(body.msg).toBe("Bad request");
                 })
             })
+            describe("/comments", () => {
+                test("200: Comments should be retrieved with the necessary properties in an array of objects", () => {
+                    return request(app)
+                    .get('/api/articles/1/comments')
+                    .expect(200)
+                    .then(({body}) => {
+                        console.log(body);
+                        expect(body.comments.length).toBe(11);
+                        body.comments.forEach((comment) => {
+                            expect(comment).toHaveProperty("comment_id", expect.any(Number));
+                            expect(comment).toHaveProperty("votes", expect.any(Number));
+                            expect(comment).toHaveProperty("created_at", expect.any(String));
+                            expect(comment).toHaveProperty("author", expect.any(String));
+                            expect(comment).toHaveProperty("body", expect.any(String));
+                            expect(comment).toHaveProperty("article_id", expect.any(Number));
+                        })
+                    })
+                })
+                test("200: Comments should be sorted by most recent first.", () => {
+                    return request(app)
+                    .get('/api/articles/1/comments')
+                    .expect(200)
+                    .then(({body}) => {
+                        expect(body.comments).toBeSortedBy("created_at", { descending: true });
+                    })
+                })
+                test("200: Article with no comments should send empty array", () => {
+                    return request(app)
+                    .get('/api/articles/10/comments')
+                    .expect(200)
+                    .then(({body}) => {
+                        expect(body.comments.length).toBe(0);
+                    })
+                })
+                test("404: Should return \"Article ID not found\" if given valid but non-existent ID", () => {
+                    return request(app)
+                    .get('/api/articles/999/comments')
+                    .expect(404)
+                    .then(({body}) => {
+                        expect(body.msg).toBe("Article ID not found")
+                    })
```

Comment:
Great testing so far! Could we also have a test for if we pass in an invalid ID? E.g. api/articles/invalid-id/comments

Reply:
This has been refactored using existing error handles.